### PR TITLE
checker: stricter `unknown type` checks, show better suggestions

### PR DESCRIPTION
**Pull request by Delta456**

### CODE REVIEW

**Comment by Delta456** [comment]
The CI Error is not related to my PR.

File: vlib/v/checker/tests/any_int_float_ban_err.out [hunk 1]
```diff
@@ -10,29 +10,31 @@ vlib/v/checker/tests/any_int_float_ban_err.vv:2:1: error: type `int_literal` doe
...
+Did you mean `Int`?
     7 | }
     8 | 
     9 | fn foo(i int_literal) int_literal {
       | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    10 |     return i
    11 | }
-vlib/v/checker/tests/any_int_float_ban_err.vv:13:1: error: unknown type `int_literal`
+vlib/v/checker/tests/any_int_float_ban_err.vv:13:1: error: unknown type `int_literal`.
+Did you mean `Int`?
```
Comment (StunxFS) [inline vlib/v/checker/tests/any_int_float_ban_err.out:37]: Why `Int`? Shouldn't it be `int`?

Comment (spaceface777) [reply vlib/v/checker/tests/any_int_float_ban_err.out:37]: I guess the compiler decided that `Int` was a more accurate suggestion than `int` ¯\_(ツ)_/¯

That is out of scope for this PR, but the fact that it stopped suggesting the `int_literal`/`float_literal` types is intentional.

**Comment by spaceface777** [comment]
> ```
> vlib\time\time_windows.c.v:53:1: error: unknown type `time.time_t`.
> Did you mean `C.time_t`?
>    51 | }
>    52 | 
>    53 | fn C._mkgmtime(&C.tm) time_t
>       | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>    54 | 
> ```
 
 :)

**Comment by StunxFS** [comment]
What I like about these modifications is that errors are always discovered that go unnoticed :D

**Comment by medvednikov** [comment]
Good one @Delta456 !

**Pull request merged**